paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -104,7 +110,7 @@ protected AbstractFileStoreTable(
             Path path,
             TableSchema tableSchema,
             CatalogEnvironment catalogEnvironment) {
-        this.fileIO = fileIO;
+        this.fileIO = new FileIOWrapper(fileIO);
```

Comment:
We may need to introduce an option to enable `MetricsFileIO`.

Reply:
Done

> We may need to introduce an option to enable `MetricsFileIO`.